### PR TITLE
Added override for set_data_from_engine

### PR DESCRIPTION
**Pull request by Peterburnett**
Functionally identical to parent, apart from a small change in functionality when files are embedded inside of blocks

### CODE REVIEW

File: classes/document.php [hunk 1]
```diff
@@ -293,6 +293,33 @@ public function export_for_webservice() {
...
+
+            // Optional params might not be there.
+            if (isset($docdata[$fieldname])) {
+                if ($field['type'] === 'tdate') {
+                    // Time fields may need a preprocessing.
+                    $this->set($fieldname, static::import_time_from_engine($docdata[$fieldname]));
+                } else {
+                    // No way we can make this work if there is any multivalue field.
+                    if (is_array($docdata[$fieldname])) {
+                        throw new \core_search\engine_exception('multivaluedfield', 'search_solr', '', $fieldname);
```
Comment (dmitriim) [inline classes/document.php]: Not going to do full review, but probably not a good idea to use a lang string from search_solr

Comment (Peterburnett) [reply classes/document.php]: That was from core, I must have missed it. This is a copy past from the parent document class, with a small change.

Comment (Peterburnett) [reply classes/document.php]: Im not sure if that would even be necessary in Elastic, I haven't seen any multivalue fields, not sure if it something the engine outputs

**Comment by mattporritt** [comment]
Hi @Peterburnett ,
I've had a look at this and I'm happy for it to be merged in.
However, it does feel like we should also look at how this content is being added to the index in the first place and if there is anything that can be done to fix this problem at index time. 
Cheers,
Matt P

**Pull request merged**